drivers/frequency/adf4371/adf4371.c

```diff
@@ -0,0 +1,820 @@
+			     ADF4371_RF_DIV_SEL_MSK,
+			     ADF4371_RF_DIV_SEL(dev->rf_div_sel));
+	if (ret < 0)
+		return ret;
+
+	/*
+	 * The optimum bleed current is set by ((4/N) × ICP)/3.75,
+	 * where ICP is the charge pump current in μA
+	 */
+	cp_bleed = DIV_ROUND_UP(400 * dev->cp_settings.icp, dev->integer * 375);
+	cp_bleed = clamp(cp_bleed, 1U, 255U);
+	ret = adf4371_write(dev, ADF4371_REG(0x26), cp_bleed);
+	if (ret < 0)
+		return ret;
+	/*
+	 * Set to 1 when in INT mode (when FRAC1 = FRAC2 = 0),
+	 * and set to 0 when in FRAC mode.
+	 */
+	if (dev->fract1 == 0 && dev->fract2 == 0)
+		int_mode = 0x01;
+
+	ret = adf4371_write(dev, ADF4371_REG(0x2B), int_mode);
+	if (ret < 0)
+		return ret;
+
+	return adf4371_write(dev, ADF4371_REG(0x10), dev->integer & 0xFF);
+}
+
+/**
+ * Power down one channel.
+ * @param dev - The device structure.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int32_t adf4371_channel_power_down(struct adf4371_dev *dev,
+		uint32_t channel,
+		bool power_down)
+{
+	uint32_t bit, reg;
+	uint8_t readval;
+	int32_t ret;
+
+	reg = adf4371_pwrdown_ch[channel].reg;
+	bit = adf4371_pwrdown_ch[channel].bit;
+	ret = adf4371_read(dev, reg, &readval);
+	if (ret < 0)
+		return ret;
+
+	readval &= ~BIT(bit);
+	readval |= (!power_down << bit);
+
+	return adf4371_write(dev, reg, readval);
+}
+
+/**
+ * Configure the channels.
+ * @param dev - The device structure.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int32_t adf4371_channel_config(struct adf4371_dev *dev)
+{
+	uint64_t rate;
+	int32_t i, ret;
+
+	for (i = 0; i < dev->num_channels; i++) {
+		if (dev->channel_cfg[i].freq == 0)
+			continue;
+
+		rate = adf4371_pll_fract_n_get_rate(dev, i);
+		if (rate == 0) {
+			ret = adf4371_set_freq(dev, dev->channel_cfg[i].freq, i);
+			if (ret < 0)
+				return ret;
+		} else if (rate != dev->channel_cfg[i].freq) {
+			printf("Clock rate for chanel %"PRId32" is not in sync\n", i);
+			return FAILURE;
+		}
+		/* Powerup channel */
+		if (dev->channel_cfg[i].enable) {
+			ret = adf4371_channel_power_down(dev, i, false);
+			if (ret < 0)
+				return ret;
+		}
+	}
+
+	return 0;
+}
+
+/**
+ * Setup the device.
+ * @param dev - The device structure.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int32_t adf4371_setup(struct adf4371_dev *dev)
+{
+	uint32_t vco_alc_timeout = 1;
+	uint32_t synth_timeout = 2;
+	uint32_t vco_band_div;
+	uint32_t timeout = 1;
+	uint32_t tmp;
+	bool en = true;
+	uint8_t mask;
+	uint8_t val;
+	int32_t ret;
+	int32_t i;
+
+	ret = adf4371_write(dev, ADF4371_REG(0x0), ADF4371_RESET_CMD);
+	if (ret < 0)
+		return ret;
+
+	if (dev->spi_3wire_en)
+		en = false;
+
+	ret = adf4371_update(dev, ADF4371_REG(0x0),
+			     ADF4371_SDO_ACT_MSK | ADF4371_SDO_ACT_R_MSK,
+			     ADF4371_SDO_ACT(en) | ADF4371_SDO_ACT_R(en));
+	if (ret < 0)
+		return ret;
+
+	for (i = 0; i < ARRAY_SIZE(adf4371_reg_defaults); i++) {
```

Comment:
I understand the need of adf4371_reg_defaults but I think the original linux driver and this driver might not use it properly. The situation where you'd need it is when you keep a cache of the device settings and sometimes you may need to read values from that cache. At driver initialization you'd need to initialize the cache with settings from adf4371_reg_defaults. But here, there is no cache and adf4371_reg_defaults is only used to write these values once, at initialization to the device. I didn't check the datasheet but I believe these are default register values after reset so this look won't do anything useful. 
It doesnt harm, but it wont help either.